### PR TITLE
doc sdk guide

### PR DESCRIPTION
**Pull request by 00k**

### CODE REVIEW

**Comment by 00k** [comment]
some updates are lost 😢   hold on please ......

**Comment by 00k** [comment]
done, please review

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ Tera is a high performance distributed NoSQL database, which is inspired by goog
 * Support RAMDISK/SSD/DFS tiered cache
 * Block cache and Bloom Filters for real-time queries
 * Multi-type table support (RAMDISK/SSD/DISK table)
-* Easy to use [C++](doc/sdk_dev_guide.md)/[Java](sdk_dev_guide_for_java.md)/[Python](sdk_dev_guide_for_python.md)/[REST-ful](doc/http_proxy.md) API for client access
+* Easy to use [C++](doc/en/sdk_guide.md)/[Java](sdk_dev_guide_for_java.md)/[Python](sdk_dev_guide_for_python.md)/[REST-ful](doc/http_proxy.md) API for client access
```
Comment (taocp) [inline README.md]: dead link

File: doc/en/sdk_guide.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+>   * tera::TableDescriptor, tera::LocalityGroupDescriptor, tera::ColumnFamilyDescriptor
+>  2. [Samples](#samples)
+>   * [Table management](#management-samples)
+>     * Table description
+>     * Table control
+>   * [Data manipulation](#manipulation-samples)
+>     * Read
+>     * Write
+>     * Scan
+>  3. [Java SDK](doc/sdk_dev_guide_for_java.md)
```
Comment (taocp) [inline doc/en/sdk_guide.md]: dead link

File: doc/en/sdk_guide.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+* Set a callback in it to activate asynchronous write.
+* Check if the write operation is successful by *RowMutation::GetError()*.
+* RowMutation supports atomic operations such as counter, string-append, put-if-absent, etc.
+
+### 1.5 tera::ScanDescriptor/tera::ResultStream --- *used to perform sequential read operations*
+
+A ScanDescriptor describes a sequential read demand, which may include range of row key, set of columns, time range, maximum and minimum versions, user defined filters, etc.
+A ResultStream is a heap-allocated iterator over the contents of the desired cells of table.
+* Obtain an instance of ScanDescriptor from *new tera::ScanDescriptor()*, apply it to server meanwhile obtain an instance of ResultStream by *tera::Table::Scan()*.
+* If ScanDescriptor or ResultStream are no longer needed, *delete* them.
```
Comment (yvxiang) [inline doc/en/sdk_guide.md]: is no longer needed?

**Comment by taocp** [comment]
Maybe you need to resolve conflicts

File: doc/en/sdk_guide.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+>   * [Table management](#management-samples)
+>     * Table description
+>     * Table control
+>   * [Data manipulation](#manipulation-samples)
+>     * Read
+>     * Write
+>     * Scan
+>  3. [Java SDK](../sdk_dev_guide_for_java.md)
+>  4. [Python SDK](../sdk_dev_guide_for_python.md)
+
```
Comment (caijieming-baidu) [inline doc/en/sdk_guide.md]: 空的？

**Comment by taocp** [comment]
build

**Pull request merged**